### DISCUSSION TITLE
Machine learning AVX support?

### DISCUSSION THREAD

**Discussion by flobernd** [immich-app/immich#12709] [Q&A] [answered]
Hi, does the Immich machine learning module utilize AVX-512 and AVX-VNNI?

The OpenVINO framework seems to be able to make use of these instruction sets, but Immich only mentions GPU support in this context.

Is CPU acceleration by use of specialized instruction sets something that works out of the box, or does Immich use OpenVINO in a way that only makes use of GPUs?

**Comment by bo0tzz** [comment]
cc @mertalev

**Comment by mertalev** [comment, marked as answer]
Hi! ONNX Runtime can use AVX-512/VNNI, yes. We run models at full precision, though. I believe quantized models take more advantage of these instructions.
